daemon/settings.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2021 Open Mobile Platform LLC.
+ *
+ * You may use this file under the terms of the BSD license as follows:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   1. Redistributions of source code must retain the above copyright
+ *      notice, this list of conditions and the following disclaimer.
+ *   2. Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer
+ *      in the documentation and/or other materials provided with the
+ *      distribution.
+ *   3. Neither the names of the copyright holders nor the names of its
+ *      contributors may be used to endorse or promote products derived
+ *      from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation
+ * are those of the authors and should not be interpreted as representing
+ * any official policies, either expressed or implied.
+ */
+
+#ifndef  SETTINGS_H_
+# define SETTINGS_H_
+
+# include <stdbool.h>
+# include <glib.h>
+
+# ifdef __cplusplus
+#  define SETTINGS_C_BEG extern "C" {
+#  define SETTINGS_C_END };
+# else
+#  define SETTINGS_C_BEG
+#  define SETTINGS_C_END
+# endif
+
+SETTINGS_C_BEG
+
+/* ========================================================================= *
+ * Types
+ * ========================================================================= */
+
+typedef struct config_t       config_t;
+typedef struct settings_t     settings_t;
+typedef struct usersettings_t usersettings_t;
+typedef struct appsettings_t  appsettings_t;
+typedef struct control_t      control_t;
+typedef struct stringset_t    stringset_t;
+
+typedef enum
+{
+    APP_ALLOWED_UNSET,
+    APP_ALLOWED_ALWAYS,
+    APP_ALLOWED_NEVER,
+    APP_ALLOWED_COUNT
+    // keep app_allowed_name[] in sync
```

Comment:
Care to elaborate. I do not follow.

(app_allowed_name[] = human readable names for enum values for diagnostic purposes, which naturally needs to be updated if new enum values are ever added)

Reply:
I mean app_allowed_t and app_agreed_t look similar to me.

Reply:
Despite the looks (now), those are logically different. So, while using one enum to cover both would be possible now, it would have negative impact on readability and it would make future changes more difficult than what they need to be. [And there has already been chatter about extending the allowed set.]